src/readwrite.jl

```diff
@@ -13,3 +13,42 @@ function readcpp(io::IO, ::Type{Vector{T}}; kwargs...) where {T}
   end
   return v
 end
+
+function HDF5.read(
+  parent::Union{HDF5.File,HDF5.Group}, name::AbstractString, ::Type{AutoType}
+)
+  type_string = try
+    @suppress_err begin
+      g = open_group(parent, name)
+      type_attribute = attributes(g)["type"]
+      HDF5.read(type_attribute)
+    end
+  catch
+    # If the file doesn't have the format
+    # expected of ITensor HDF5 files, return `nothing`
+    return nothing
```

Comment:
One question I would have is whether the functionality in line 40 below (`isnothing(r) && return HDF5.read(parent, name)`) should just be put here in this overload of HDF5.read?

Reply:
Then both this `HDF5.read` overload and `ITensors.read` would have this fallback behavior.

Reply:
Good suggestion, I can change that.